### ISSUE TITLE
Drop use-download-and-update-url eslint rule

### ISSUE THREAD

**Issue by jesus2099** [Yash-Singh1/eslint-plugin-userscripts#79]
This [`use-download-and-update-url`](https://github.com/Yash-Singh1/eslint-plugin-userscripts/blob/main/docs/rules/use-download-and-update-url.md) says that, if you have a `downloadURL` then you should also have a (redundant) `updateURL`.

Violentmonkey, [when no `updateURL`, will logically use `downloadURL` (fallback to URL where the script was installed from)](https://github.com/violentmonkey/violentmonkey/blob/0fd1a0345ed95f76efabce86e94abd7273dc5b96/src/common/index.js#L226-L227) to both check version updates and download updates.

I guess it's also the case for Greasemonkey and Tampermonkey (not open source), isn't it?

I did not see reference as to for what userscript extension(s) with different behaviour this rule was written for:

Actually `updateURL` is no longer **officially** supported in Violentmonkey and in Greasemonkey.
And even `downloadURL` is apprently no longer **officially** supported in Greasemonkey.

https://violentmonkey.github.io/api/metadata-block/

https://wiki.greasespot.net/Metadata_Block
https://wiki.greasespot.net/index.php?title=Metadata_Block&diff=7489&oldid=7469

https://www.tampermonkey.net/documentation.php

**Comment by double-beep** [comment]
Notice the [description](https://yash-singh1.github.io/eslint-plugin-userscripts/#/rules/use-download-and-update-url) of the rule is (emphasis mine):

> The `use-download-and-update-url` rule verifies that if the `updateURL` attribute is present then the `downloadURL` attribute is too **and vice versa**.

---

> Actually `updateURL` is no longer **officially** supported in Violentmonkey and in Greasemonkey.
> And even `downloadURL` is apprently no longer **officially** supported in Greasemonkey.

This rule was created precisely to deal with such compatibility issues between different userscript managers.



<sub>Also: Greasemonkey is [open-source](https://github.com/greasemonkey/greasemonkey).</sub>

**Comment by jesus2099** [comment]
If we follow this rule we make code redundancy when we don't need a different `updateURL` than `downloadURL` (which is 99.99% of the time, and a risk of forgetting to update one or the other.

I have not found any userscript extension that does not work well with `dowloadURL` and no `updateURL`.

Have you ever seen one?
If not, this should be **allowed**: `dowloadURL` without `updateURL`

You can **keep** the **vice versa** rule, though, it seems wise.

Also my remark some docs not mentioning these URL anymore means that this rule should even less advocate to add a redundant `updateURL`. ;)

---

Yes, I know only Tampermonkey is not open source. :)
And Greasemonkey is the original extension, that deserves biggest respect!

| Extension     | Open source | Multi-browser |
|---------------|-------------|---------------|
| Greasemonkey  | Yes         | No            |
| Tampermonkey  | No          | Yes           |
| Violentmonkey | Yes         | Yes           |

**Comment by Yash-Singh1** [comment]
Ok, so I did a bit more research into this and it seems as if the original purpose of having a different `updateURL` and `downloadURL` was a bit more complicated than an alias. Apparently, `updateURL` was intended to be a URL pointing to a versioning "metadata" file for your userscript. People would use this if their userscript is extremely large, so it would make sense to separate the versioning info into a separate file to save bandwith. As far as I knew, I thought that `updateURL` and `downloadURL` act as aliases to each other in most implementations.

> I have not found any userscript extension that does not work well with `downloadURL` and no `updateURL`.

:+1: This seems to hold for pretty much every userscript manager I came across, even the minor ones.

> It's the vice versa rule that should be dropped, IMO

This seems like the right way to go, as I have seen userscript managers that do not respect `updateURL` for source downloads.
